components/mediation-connector/org.wso2.carbon.connector.core/src/main/java/org/wso2/carbon/connector/core/RefreshAccessToken.java

```diff
@@ -0,0 +1,175 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.connector.core;
+
+import org.apache.axiom.om.OMText;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseLog;
+import org.apache.synapse.config.Entry;
+import org.apache.synapse.registry.Registry;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * This class can be used by connectors to refresh OAuth 2.0 access tokens by setting the following mandatory
+ * properties in message context: uri.var.hostName, uri.var.refreshToken. By default this class constructs the refresh
+ * url in the format "{uri.var.hostName}/services/oauth2/token?grant_type=refresh_token&client_id=
+ * {uri.var.clientId}&client_secret={uri.var.clientSecret}&refresh_token={uri.var.refreshToken}&format=json".
+ * Here client_id and client_secret are optional. If you want to use a different url please set the custom url to
+ * uri.var.customRefreshUrl in message context prior to using this class mediator.
+ *
+ * After refresh call this will set the uri.var.accessToken, and uri.var.apiUrl in the message context to be used by
+ * subsequent calls.
+ */
+public class RefreshAccessToken extends AbstractConnector {
+    private CloseableHttpClient httpClient = HttpClients.createDefault();
+    protected final String PROPERTY_PREFIX = "uri.var.";
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        Registry registry = messageContext.getConfiguration().getRegistry();
+        String accessTokenRegistryPath = (String) messageContext.getProperty(PROPERTY_PREFIX +
+                "accessTokenRegistryPath");
+        handleRefresh(messageContext, registry, accessTokenRegistryPath);
+    }
+
+    protected boolean handleSavedCredentialReuse (MessageContext messageContext, Registry registry,
```

Comment:
Please add a method comment and rename this method name also to be more meaningful

Reply:
need to reflect what is being returned in this method